gensim/models/word2vec.py

```diff
@@ -520,7 +521,12 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
         total_words = 0
         min_reduce = 1
         vocab = defaultdict(int)
+        checked_string_types = 0
         for sentence_no, sentence in enumerate(sentences):
+            if not checked_string_types:
+                if isinstance(sentence, string_types):
+                    logger.warn("'sentences' should be a list of list of unicode strings. %s provided.", type(sentence))
```

Comment:
To be pedantically accurate, `sentences` need not be a list – just an iterable. But the important thing being checked here is whether the first item is the expected list-of-strings. So I would word this warning as: "Each `sentences` item should be a list of words (usually unicode strings). First item here is instead plain %s." The warning in doc2vec could be similarly: "Each `words` should be a list of words (usually unicode strings). First 'words' here is instead plain %s".) (I find that clearer than just "%s provided" which may at first glance seem like no contradiction with preceding advice.)